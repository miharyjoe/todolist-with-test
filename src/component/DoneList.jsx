import React from "react";

const DoneList = ({ todos }) => (
  <ul className="liste">
    {todos.map((todo, index) => (
      <li key={index}>{todo}</li>
    ))}
  </ul>
);

export default DoneList;
