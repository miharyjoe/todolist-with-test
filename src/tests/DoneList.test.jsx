import React from "react";
import { render } from "@testing-library/react";
import DoneList from "../component/DoneList";

describe("DoneList", () => {
  it("snapshot: renders done task", () => {
    const todos = ["Todo 1", "Todo 2", "Todo 3"];
    const { container } = render(<DoneList todos={todos} />);
    expect(container).toMatchSnapshot();
  });

  it("renders a list of done task", () => {
    const todos = ["Todo 1", "Todo 2", "Todo 3"];
    const { getByText } = render(<DoneList todos={todos} />);
    todos.forEach((todo) => {
      expect(getByText(todo)).toBeInTheDocument();
    });
  });
});
