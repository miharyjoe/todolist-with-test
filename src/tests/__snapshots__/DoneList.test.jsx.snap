// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneList snapshot: renders done task 1`] = `
<div>
  <ul
    class="liste"
  >
    <li>
      Todo 1
    </li>
    <li>
      Todo 2
    </li>
    <li>
      Todo 3
    </li>
  </ul>
</div>
`;
